src/tetris/field/field.ts

```diff
@@ -118,51 +130,55 @@ export default class Field {
 	}
 
 	private _setupField(): void {
-		const iterator = this._blocks.keys();
+		const iterator = this._blockRows.keys();
 		for (let key of iterator) {
-			this._blocks[key] = new Array(this._width).fill(null);
+			this._blockRows[key] = new Array(this._width).fill(null);
 		}
 	}
 
-	private _addToField(blocks: Block[], brickOffset: Vector2): void {
+	private _addToField(blocks: Block[]): void {
 		blocks.forEach(block => {
-			this._blocks[block.currentPosition.x][block.currentPosition.y] = block;
+			if (this.isPositionOutOfBounds(block.currentPosition, true)) {
+				return;
+			}
+			this._blockRows[block.currentPosition.y][block.currentPosition.x] = block;
 		});
 	}
 
 	private _deleteCompletedRows(): number {
 		let rowsDeleted = 0;
-		for (let y = 0; y < this.height; y++) {
-			let rowCompleted = true;
-			for (let x = 0; x < this.width; x++) {
-				if (!this._blocks[x][y]) {
-					rowCompleted = false;
-					break;
-				}
-			}
+		this._blockRows.forEach((row, rowIndex) => {
+			let rowCompleted = row.every(block => block !== null);
```

Comment:
`block => block !== null` could be simplified to `block => !!block`

Reply:
I like this variant more, its more explicit.